pallets/utxo/tokens/src/lib.rs

```diff
@@ -5,24 +5,33 @@ use serde::{Deserialize, Serialize};
 
 use codec::{Decode, Encode};
 use frame_support::{dispatch::Vec, RuntimeDebug};
+use sp_core::{sr25519::Public, H256};
 
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Hash)]
-pub struct TokenInstance {
-    pub id: u64,
-    pub name: Vec<u8>,
-    pub ticker: Vec<u8>,
-    pub supply: u128,
-    // We can add another fields like:
-    //      pub number_format: NumberFormat,
-    //      pub image: UUID,
-    //      pub transaction: XXX,
+pub enum TokenInstance {
+    Normal {
+        id: H256,
+        name: Vec<u8>,
+        ticker: Vec<u8>,
+        supply: u128,
+        // We can add another fields like:
+        //      pub number_format: NumberFormat,
+        //      pub image: UUID,
+        //      pub transaction: XXX,
+    },
+    Nft {
+        id: H256,
+        data_hash: [u8; 32],
+        data_url: Vec<u8>,
+        creator_pubkey: [u8; 32],
```

Comment:
There's no guarantee of a 32 byte pubkey

Reply:
This might be the time to serialise all of this, then we don't need to worry about pk size or extra data fields for different token types etc...

Reply:
Agreed. Same for hashes. Perhaps using some `impl` for an interface for such components could be a good idea.

Reply:
To be honest, I don't see any reason for serialization this place instead of having just enum type over here. Deriving Encode and Decode above the struct have already made serialization for the whole structure. 

But, if it might help later then okay. Done